catalog/plugin/catalog-plugin-async/src/main/java/org/codice/ddf/catalog/plugin/async/api/internal/ProcessUpdateItem.java

```diff
@@ -0,0 +1,55 @@
+/**
+ * Copyright (c) Codice Foundation
+ * <p>
+ * This is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser
+ * General Public License as published by the Free Software Foundation, either version 3 of the
+ * License, or any later version.
+ * <p>
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.catalog.plugin.async.api.internal;
+
+import ddf.catalog.data.Metacard;
+
+/**
+ * <p>
+ * <b> This code is experimental. While this interface is functional and tested, it may change or be
+ * removed in a future version of the library. </b>
+ * </p>
+ *
+ * The {@code ProcessUpdateItem} represents the data that will be processed by the {@link ProcessingFramework}
+ * when handling a {@link ProcessUpdateRequest}.
+ */
+public interface ProcessUpdateItem {
+
+    /**
+     * @return the original {@code Metacard}
+     */
+    Metacard getOldMetacard();
+
+    /**
+     * @return the updated {@code Metacard}
+     */
+    Metacard getNewMetacard();
+
+    /**
+     * @return the original resource
+     */
+    ProcessResource getProcessResource();
+
+    /**
+     * Determines if the new {@code Metacard} was modified during processing.
+     *
+     * @return {@code true} if modified, {@code false} otherwise
+     */
+    boolean isNewMetacardModified();
```

Comment:
I would suggest making this consistent with `ProcessResource` and rename these functions to `isModifed` and `markAsModified` for consistency.  The JavaDoc makes it clear that the state of the new metacard is the state being tracked for this object.